s3_to_socrata.py

```diff
@@ -0,0 +1,177 @@
+#!/usr/bin/env python3
+"""
+Download finance json files and publish them to Socrata
+"""
+import argparse
+from datetime import datetime, timezone, timedelta
+import json
+import logging
+import os
+
+import boto3
+import sodapy
+import pandas as pd
+
+import utils
+
+AWS_ACCESS_ID = os.getenv("AWS_ACCESS_ID")
+AWS_PASS = os.getenv("AWS_PASS")
+BUCKET_NAME = os.getenv("BUCKET_NAME")
+
+SO_WEB = os.getenv("SO_WEB")
+SO_TOKEN = os.getenv("SO_TOKEN")
+SO_USER = os.getenv("SO_USER")
+SO_PASS = os.getenv("SO_PASS")
+
+TASK_DATASET = os.getenv("TASK_DATASET")
+DEPT_UNITS_DATASET = os.getenv("DEPT_UNITS_DATASET")
+
+
+def get_socrata_client():
+    return sodapy.Socrata(
+        SO_WEB, SO_TOKEN, username=SO_USER, password=SO_PASS, timeout=60,
+    )
+
+
+def aws_list_files(client):
+    """
+    Returns a list of files in an S3 bucket.
+    :return: object
+    """
+    response = client.list_objects(Bucket=BUCKET_NAME,)
+
+    file_list = []
+    for content in response.get("Contents", []):
+        file_list.append(content.get("Key"))
+
+    return file_list
+
+
+def get_dept_unit(client, socrata_client):
+    """
+    Gets the units.json file and sends the data to socrata 
+
+    Parameters
+    ----------
+    client : AWS Client object
+    
+    socrata_client : Socrata client object
+
+    Returns
+    -------
+    None.
+
+    """
+    response = client.get_object(Bucket=BUCKET_NAME, Key="units.json")
+    data = json.load(response.get("Body"))
+    socrata_client.upsert(DEPT_UNITS_DATASET, data)
+    logger.debug("Sent units data to Socrata")
+
+
+def get_task_orders(client, socrata_client):
+    """
+    Gets the task_orders.json file and sends the data to socrata 
+
+    Parameters
+    ----------
+    client : AWS Client object
+        
+    socrata_client : Socrata client object
+
+    Returns
+    -------
+    None.
+
+    """
+    response = client.get_object(Bucket=BUCKET_NAME, Key="task_orders.json")
+
+    df = pd.read_json(response.get("Body"), orient="records")
+    # Transforms the file to fit the Socrata dataset
+    data = transform_records(df)
+    socrata_client.upsert(TASK_DATASET, data)
+    logger.debug("Sent task data to Socrata")
+
+
+def transform_records(df):
+    """
+    Transforms the task_orders.json file to be in the same format as the socrata dataset
+
+    Parameters
+    ----------
+    df : Pandas Dataframe
+        Raw JSON file from S3.
+
+    Returns
+    -------
+    Dict
+        Transformed file.
+
+    """
+
+    df = df.rename(
```

Comment:
where the field mapping happens between the task_orders.json and the socrata dataset